lib/ProductOpener/DataQualityFood.pm

```diff
@@ -591,22 +591,59 @@ sub check_nutrition_data_energy_computation ($product_ref) {
 				$computed_energy += $grams * $energy_per_gram;
 			}
 
-			# Compare to specified energy value with a tolerance of 30% + an additiontal tolerance of 5
-			if (   ($computed_energy < ($specified_energy * 0.7 - 5))
-				or ($computed_energy > ($specified_energy * 1.3 + 5)))
-			{
-				# we have a quality problem
-				push @{$product_ref->{data_quality_errors_tags}},
-					"en:energy-value-in-$unit-does-not-match-value-computed-from-other-nutrients";
+			# following error/warning should be ignored for some categories
+			# for example, lemon juices containing organic acid, it is forbidden to display organic acid in nutrition tables but
+			# organic acid contributes to the total energy calculation
+			my $skip_energy_error = 0;
+			if (defined $product_ref->{categories_tags}) {
+				# start from end, categories_tags has the most specific categories at the end
+				my $i = @{$product_ref->{categories_tags}} - 1;
+				while (
+					($i >= 0)
+					and (
+						not(
+							# category - or a parent - with expected tag
+							defined get_inherited_property(
+								"categories", $product_ref->{categories_tags}[$i],
+								"ignore_energy_calculated_error:en"
+							)
+							and (
+								# we expect single letter a, b, c, d, e for nutriscore grade in the taxonomy. Case insensitive (/i).
+								get_inherited_property(
+									"categories", $product_ref->{categories_tags}[$i],
+									"ignore_energy_calculated_error:en"
+								) =~ /yes$/i
+							)
+						)
+					)
+					)
+				{
+					$i--;
+				}
+
+				if ($i >= 0) {
+					$skip_energy_error = 1;
+				}
 			}
```

Comment:
Can we make it a separate method ? (easier to read code)

Reply:
There is a new function/subroutine in **Tags.pm** (because it is based on **get_inherited_property** subroutine).
That new subroutine is called from **DataQualityFood.pm**

The idea would be to reuse this function in the other PR (https://github.com/openfoodfacts/openfoodfacts-server/pull/8360)
(eventually) It could be improved, because we expect a "yes" and we could maybe add what is expected as input parameter in the function. Here we expect a "yes", in the other PR we expect a nutriscore. What do you think?